### PR TITLE
Add support CI_SCAN for a link.

### PR DESCRIPTION
**Pull request by amine3**
This PR adds support in the CLI of the CI_SCAN for the asset type link.

Command example: `ostorlab --api-key APIKEY ci-scan run --title test1 --api-schema /tmp/schema.txt --scan-profile=full_web_scan link --url https://google.com`

Output:

image

### CODE REVIEW

File: src/ostorlab/apis/scan_create.py [hunk 1]
```diff
@@ -99,3 +101,117 @@ def files(self) -> Optional[Dict]:
...
+class CreateWebScanAPIRequest(request.APIRequest):
+    """Create web scan API from a file."""
+
+    def __init__(
+        self,
+        title: str,
+        urls: [List[str]],
+        scan_profile: str,
+        sboms: list[io.FileIO] = None,
+        api_schema: io.FileIO = None,
```
Comment (rabsondev) [inline src/ostorlab/apis/scan_create.py:115]:
```suggestion
        sboms: Optional[list[io.FileIO]] = None,
        api_schema: Optional[io.FileIO] = None,
```

File: src/ostorlab/apis/scan_create.py [hunk 2]
```diff
@@ -99,3 +101,117 @@ def files(self) -> Optional[Dict]:
...
+        proxy: Optional[str] = None,
+        qps: Optional[int] = None,
+        filtered_url_regexes: [List[str]] = None,
```
Comment (rabsondev) [inline src/ostorlab/apis/scan_create.py:118]:
```suggestion
        filtered_url_regexes: Optional[[List[str]]] = None,
```

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        test_credentials_role,
+        test_credentials_url,
+    ):
+        credentials.append(
+            test_credentials_create_api.TestCredentialLogin(
+                login=login, password=password, role=role, url=url
+            )
+        )
+
+    if test_credentials_name:
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:40]: Explicit check.

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            )
+        )
+
+    return credentials
+
+
+def run_link_scan(ctx: click.core.Context, url: List[str]) -> None:
+    """Create scan for links."""
+    ci_logger = ctx.obj["ci_logger"]
+    if ctx.obj.get("api_key"):
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:55]:
```suggestion
    if ctx.obj.get("api_key") is not None:
```

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        api_schema = ctx.obj["api_schema"]
+        filtered_url_regexes = ctx.obj["filtered_url_regexes"]
+        proxy = ctx.obj["proxy"]
+        qps = ctx.obj["qps"]
+        runner = authenticated_runner.AuthenticatedAPIRunner(
+            api_key=ctx.obj.get("api_key")
+        )
+        try:
+            test_credentials = _prepare_test_credentials(ctx)
+            if test_credentials:
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:70]: Explicit check.

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                title,
+                scan_profile,
+                url,
+                credential_ids,
+                runner,
+                sboms,
+                api_schema,
+                filtered_url_regexes,
+                proxy,
+                qps,
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:91]: Please use `kwargs`.

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            filtered_url_regexes=filtered_url_regexes,
+            proxy=proxy,
+            qps=qps,
+        )
+    )
+    scan_id = scan_result.get("data").get("createWebScan").get("scan").get("id")
+    return scan_id
+
+
+def _create_test_credentials(test_credentials, runner):
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:142]: Missing type hints for params and return type.

File: src/ostorlab/cli/ci_scan/run/assets/link.py [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+            scan_profile=scan_profile,
+            test_credential_ids=credential_ids,
+            sboms=sboms,
+            api_schema=api_schema,
+            filtered_url_regexes=filtered_url_regexes,
+            proxy=proxy,
+            qps=qps,
+        )
+    )
+    scan_id = scan_result.get("data").get("createWebScan").get("scan").get("id")
```
Comment (rabsondev) [inline src/ostorlab/cli/ci_scan/run/assets/link.py:138]: What if one of the `.get` returns `None`.

**Review by rabsondev — CHANGES_REQUESTED** [review]

File: src/ostorlab/apis/scan_create.py [hunk 9]
```diff
@@ -10,9 +10,11 @@
 SCAN_PROFILES = {
     "fast_scan": "Fast Scan",
     "full_scan": "Full Scan",
+    "full_web_scan": "Full Web Scan",
```
Comment (3asm) [inline src/ostorlab/apis/scan_create.py:13]: Shouldn't we change the other to mobile scan?

File: src/ostorlab/cli/ci_scan/run/run.py [hunk 10]
```diff
@@ -99,6 +99,34 @@
...
+    required=False,
+    default=None,
+)
+@click.option(
+    "--qps",
+    "qps",
+    help="Max QPS to use if defined.",
+    required=False,
+    default=None,
+)
```
Comment (3asm) [inline src/ostorlab/cli/ci_scan/run/run.py:129]: I don't see SBOM?

**Review by 3asm — APPROVED** [review]

File: src/ostorlab/cli/ci_scan/run/run.py (hunk 10 above)
Comment (amine3) [reply src/ostorlab/cli/ci_scan/run/run.py:129]: It is already there line 93.

**Comment by oussamaessaji** [comment]
Preparing review...

**Pull request merged**